src/stirling/core/unit_connector.h

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "src/common/base/statusor.h"
+#include "src/stirling/core/connector_context.h"
+#include "src/stirling/core/data_tables.h"
+#include "src/stirling/core/frequency_manager.h"
+
+DEFINE_string(pids, "", "PIDs to profile, e.g. -pids 132,133. All processes profiled by default.");
+
+namespace px {
+namespace stirling {
+
+template <typename T>
+class UnitConnector {
```

Comment:
It seems like this is only used for testing? And only planned to be used for testing + dev binaries? So should it live in a testing folder instead of in `stirling/core`?